### DISCUSSION TITLE
await import causing error

### DISCUSSION THREAD

**Discussion by msuntharesan** [vercel/next.js#15578] [Help] [answered]
Hello,
I am trying out next 9.5 and running into an issue with top level `await import()`
Ex:
`index.tsx`
```ts
const { myFunc } = await import("./file");

export default function Main() {
  const content = myFunc();
  return <div>content</div>;
}
```

This throws
```
The default export is not a React Component in page: "/"
```
In my next.js file, I have
```
config.experiments = {
  topLevelAwait: true,
  importAsync: true,
  asyncWebAssembly: true,
  importAwait: true,
};
```

Versions:
```
"next": "9.5.0",
"typescript": "beta"

"resolutions": {
  "webpack": "^5.0.0-beta.22"
}
```

Any help is appriciated.

Thanks

**Comment by ghost** [comment]
Hello! I wanna know as to what `./file` means. Is it `file.js` or `file.jsx`?

**Reply by msuntharesan** [reply]
it is `file.ts`. it is exporting a function.

**Reply by ghost** [reply]
Ok. So, can you show me the code of `file.ts` (If you don't mind)?

**Comment by todortotev** [comment]
https://github.com/vercel/next.js/discussions/11185

**Comment by msuntharesan** [comment]
Here is an example of what I am trying to do.
https://github.com/msuntharesan/next-js-wasm

When running with a standalone webpack, this works. I will try to add that to the example as well.
Thanks.

**Reply by msuntharesan** [reply]
But the `async import` doesn't have to just wasm. Any top level `async import` throws the same error

**Reply by ghost** [reply]
Have you followed the steps in #11185 ? I hope that you have also tried something like this:
```ts
(async () => {
    var { myFunc } = await import('./file.ts'); // Make sure you have used await inside an asynchronous function
    // ....
})();
```

**Comment by msuntharesan** [comment, marked as answer]
I used react `Context` to solve this.
Use `useEffect` to load wasm module and set it as value to `Provider` then using `useContext` to get the wasm functions in components.

**Reply by ghost** [reply]
Oh wow! Alright then! Good answer. I didn't even know what `useEffect` was used for all this while y know. Now I know thx to this answer
